src/tools/PrizeTierController/hooks/useAllPrizeTierHistoryData.ts

```diff
@@ -0,0 +1,30 @@
+import { usePrizePools } from '@hooks/usePrizePools'
+import { usePrizeTierHistoryData } from '@prizeTierController/hooks/usePrizeTierHistoryData'
+import { PrizeTierConfig } from '@pooltogether/v4-utils-js'
+
+export const useAllPrizeTierHistoryData = () => {
+  const prizePools = usePrizePools()
+  const allPrizeTierHistoryData: {
+    [chainId: number]: { [prizeTierHistoryAddress: string]: PrizeTierConfig }
+  } = {}
+  let allFetched = true
+  prizePools.forEach((prizePool) => {
```

Comment:
👀 https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

This is a sneaky one. It's a _little_ awkward since `prizePools` are consistent so it doesn't immediately cause issues. But regardless, you can't call hooks from inside loops! Best solution is to expose the data fetching function and a function to generate the unique key for that hook from `usePrizeTierHistoryData` and then use `useQueries` to loop over prize pools and call that data fetching function to get an array of results. Then combine once they've all returned.

Reply:
Fixed in f0f6535
Thanks for the help :)